src/k8s/pkg/k8sd/types/feature_status.go

```diff
@@ -0,0 +1,37 @@
+package types
+
+import (
+	"time"
+
+	apiv1 "github.com/canonical/k8s/api/v1"
+)
+
+// FeatureStatus encapsulates the deployment status of a feature.
+type FeatureStatus struct {
+	// Enabled shows whether or not the deployment of manifests for a status was successful.
+	Enabled bool
+	// Message contains information about the status of a feature. It is only supposed to be human readable and informative and should not be programmatically parsed.
+	Message string
+	// Version shows the version of the deployed feature.
+	Version string
+	// UpdatedAt shows when the last update was done.
+	UpdatedAt time.Time
+}
+
+func (f FeatureStatus) ToAPI() (apiv1.FeatureStatus, error) {
+	return apiv1.FeatureStatus{
+		Enabled:   f.Enabled,
+		Message:   f.Message,
+		Version:   f.Version,
+		UpdatedAt: f.UpdatedAt,
+	}, nil
+}
+
+func FeatureStatusFromAPI(apiFS apiv1.FeatureStatus) (FeatureStatus, error) {
+	return FeatureStatus{
+		Enabled:   apiFS.Enabled,
+		Message:   apiFS.Message,
+		Version:   apiFS.Version,
+		UpdatedAt: apiFS.UpdatedAt,
+	}, nil
+}
```

Comment:
these cannot fail, remove the optional error (also helps with the `.ToAPI()` conversions in the cluster status RPC

Reply:
ah sorry saw the one above first, ok